### PR TITLE
Move MaxValidUntilBlockIncrement & MaxTraceableBlocks to protocol

### PR DESCRIPTION
**Pull request by Qiao-Jin**
Close https://github.com/neo-project/neo/pull/2028#issuecomment-718460545 #2026 #2027

### CODE REVIEW

File: src/neo/Wallets/Wallet.cs [hunk 1]
```diff
@@ -341,7 +341,7 @@ private Transaction MakeTransaction(StoreView snapshot, byte[] script, Signer[]
                     Version = 0,
                     Nonce = (uint)rand.Next(),
                     Script = script,
-                    ValidUntilBlock = snapshot.Height + Transaction.MaxValidUntilBlockIncrement,
+                    ValidUntilBlock = snapshot.Height + ProtocolSettings.Default.MaxValidUntilBlockIncrement,
```
Comment (shargon) [inline src/neo/Wallets/Wallet.cs:344]: Not related to this PR, should we use the last known height instead of our height?

**Review by shargon — DISMISSED** [review]

File: src/neo/Wallets/Wallet.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo/Wallets/Wallet.cs:344]: Maybe the wallet should not allow the user to send transactions before the blocks are fully synchronized.

**Comment by erikzhang** [comment]
We only agreed to move `MaxTraceableBlocks`.

**Comment by vncoelho** [comment]
I think moving `MaxValidUntilBlockIncrement` sounds good if just accessing the storage locally, we can cache this value every block persist. It will not change much but looks to be a good practice.

**Comment by Qiao-Jin** [comment]
> We only agreed to move `MaxTraceableBlocks`.

Have Resumed Transaction.MaxValidUntilBlockIncrement

**Review by shargon — APPROVED** [review]

**Comment by Qiao-Jin** [comment]
Can we merge this one?

**Review by erikzhang — APPROVED** [review]

**Pull request merged**